### PR TITLE
Adding physicalDepth to MediaQueryData & TestWindow

### LINKED ISSUE #33854: MediaQuery should know about Z bounds
`MediaQuery` should expose the max elevation reported to the `Window`.
~Blocked on https://github.com/flutter/flutter/issues/33853~

### PR DESCRIPTION
**Pull request by Piinks**
## Description

Following up on https://github.com/flutter/engine/pull/10414, adds the `physicalDepth` attribute to both `MediaQueryData` and `TestWindow` classes.

Also, went through affected files and updated typos and style guide nits. 

## Related Issues

Fixes #33854 

## Tests

Updated
- MediaQueryData is sane
- MediaQueryData.copyWith copies specified values
- TestWindow can fake physical depth

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by dnfield** [comment]
Can we update the TestWindow class to have this property as well?  in packages/flutter_test/lib/src/window.dart

**Comment by Piinks** [comment]
> Can we update the TestWindow class to have this property as well? in packages/flutter_test/lib/src/window.dart

Yes! I was looking for MockWindow. 🤦‍♀  Thank you!

File: packages/flutter_test/lib/src/window.dart [hunk 1]
```diff
@@ -70,43 +73,65 @@ class TestWindow implements Window {
...
     _viewInsetsTestValue = null;
     onMetricsChanged();
   }
 
-  // TODO(dnfield): Remove this ignore once custom embedders have had time to catch up
-  // And make this property actually wrap _window.viewPadding.
+  // TODO(dnfield): Remove this ignore once custom embedders have had time to
+  //  catch up and make this property actually wrap _window.viewPadding.
   // @override
   // ignore: annotate_overrides, public_member_api_docs
```
Comment (dnfield) [inline packages/flutter_test/lib/src/window.dart]: Can we remove the TODO and the ignore here, and uncomment the override?  This should be all set.

**Review by dnfield — APPROVED** [review]
LGTM with small request while you're in here.

File: packages/flutter_test/lib/src/window.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter_test/lib/src/window.dart]: Sure thing!

**Pull request merged**